network/wifi/wifi.c

```diff
@@ -48,28 +48,47 @@
 #include "error.h"
 #include "util.h"
 
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define NB_SOCKETS	MAX_CONNECTIONS
+
 /******************************************************************************/
 /*************************** Types Declarations *******************************/
 /******************************************************************************/
 
 /* Structure storing data used by a socket */
 struct socket_desc {
-	/* Buffer submitted to the at parser */
-	uint8_t			*buff;
+	/* Buffer given to at_parser */
+	struct circular_buffer	*cb;
+	/* Circula buffer size */
+	uint32_t		cb_size;
 	/* Socket type */
 	enum socket_protocol	type;
-	/* True if a socket connect was called on this id */
-	bool			connected;
+	/* Connection id */
+	uint32_t		conn_id;
+	/* States of a socket structure */
+	enum {
+		/* The socket structure is unused */
+		SOCKET_UNUSED,
+		/* Socket structure have been initialized */
+		SOCKET_DISCONNECTED,
+		/* Socket connected to a server or to a client */
+		SOCKET_CONNECTED
+	}			state;
 };
 
 /* Wifi descriptor */
 struct wifi_desc {
 	/* Sockets */
-	struct socket_desc		sockets[MAX_CONNECTIONS];
+	struct socket_desc		sockets[NB_SOCKETS];
 	/* Reference to the AT parser */
 	struct at_desc			*at;
 	/* Network interface */
 	struct network_interface	interface;
+	/* Will be used in callback */
+	int32_t				conn_id_to_sock_id[MAX_CONNECTIONS];
```

Comment:
would this need to be NB_SOCKETS? Also what is the reason to have two defines for the same thing ?
```
#define NB_SOCKETS	MAX_CONNECTIONS

Reply:
Yes, in the next PR NB_SOCKETS will be MAX_CONNECTIONS + 1 (the +1 will be used as server socket)
I do all changes at once, and when I tried to split in different commits I removed just the +1. In order to avoid extra work

Reply:
I can keep NB_SOCKETS for this commit is really needed

Reply:
Removed macro MAX_CONNECTIONS